buildkite/provider.go

```diff
@@ -83,6 +83,26 @@ func (tf *terraformProvider) Configure(ctx context.Context, req provider.Configu
 	resp.DataSourceData = client
 }
 
+func userAgent(name, providerVersion, tfVersion string) string {
+	ua := fmt.Sprintf("Terraform/%s (+https://www.terraform.io)", tfVersion)
+	if name != "" {
+		ua += " " + name
+		if providerVersion != "" {
+			ua += "/" + providerVersion
+		}
+	}
+
+	if add := os.Getenv("TF_APPEND_USER_AGENT"); add != "" {
```

Comment:
I'm not sure it's clear what `add` is here? Could it be given a clearer name? I'm thinking if someone else were to pick this up, could they immediately know what would happen if they changed/removed `add`?

Reply:
Good point this is copied from the SDKv2 UserAgent function verbatim so that we can remove the sdkv2 dependency 🙌

Reply:
@mcncl , I've modified the function with more descriptive var names